app/src/main/java/com/alphawallet/app/ui/widget/holder/TokenHolder.java

```diff
@@ -260,24 +255,15 @@ private void displayTokenIcon()
         }
         else
         {
-            String correctedAddr = Keys.toChecksumAddress(token.getAddress());
-            String tURL = assetDefinition.getTokenImageUrl(token.tokenInfo.chainId, token.getAddress());
-            if (TextUtils.isEmpty(tURL))
-            {
-                tURL = TRUST_ICON_REPO.replace(ICON_REPO_ADDRESS_TOKEN, correctedAddr);
-            }
-
-            boolean onlyTryCache = iconCheck.containsKey(correctedAddr);
+            IconItem iconItem = assetDefinition.fetchIconForToken(token);
 
             Glide.with(getContext().getApplicationContext())
```

Comment:
If we are going to use a new class for the iconItem then how about abstracting the code in .signature(...) out into the new class?